instrumentation/jmx-metrics/library/src/main/resources/jmx/rules/hadoop.yaml

```diff
@@ -0,0 +1,71 @@
+---
+rules:
+  - bean: Hadoop:service=NameNode,name=FSNamesystem
+    prefix: hadoop.dfs.
+    metricAttribute:
+      hadoop.node.name: beanattr(tag\.Hostname)
+    mapping:
+      # hadoop.dfs.capacity
+      CapacityTotal:
+        metric: capacity
+        type: updowncounter
+        unit: By
+        desc: Current raw capacity of DataNodes.
+      # hadoop.dfs.capacity.used
+      CapacityUsed:
+        metric: capacity.used
+        type: updowncounter
+        unit: By
+        desc: Current used capacity across all DataNodes.
+      # hadoop.dfs.block.count
+      BlocksTotal:
+        metric: block.count
+        type: updowncounter
+        unit: "{block}"
+        desc: Current number of allocated blocks in the system.
+      # hadoop.dfs.block.missing
+      MissingBlocks:
+        metric: block.missing
+        type: updowncounter
+        unit: "{block}"
+        desc: Current number of missing blocks.
+      # hadoop.dfs.block.corrupt
+      CorruptBlocks:
+        metric: block.corrupt
+        type: updowncounter
+        unit: "{block}"
+        desc: Current number of blocks with corrupt replicas.
+      # hadoop.dfs.volume.failure.count
+      VolumeFailuresTotal:
+        metric: volume.failure.count
```

Comment:
We could map `EstimatedCapacityLostTotal` attribute in the future to `volume.failure.capacity` to provide an estimated capacity lost due to volume failures, so the `.count` suffix is relevant to keep here.

Also, I think this is safe to keep the `volume` wording here for clarity as there are lots of different possible failure types.